alkira/resource_alkira_connector_gcp_interconnect.go

```diff
@@ -0,0 +1,341 @@
+							Optional:    true,
+							Type:        schema.TypeBool,
+						},
+						"customer_gateways": {
+							Description: "The customer gateway associated with the segment.",
+							Type:        schema.TypeList,
+							Required:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"loopback_ip": {
+										Description: "The customer gateway IP address " +
+											"which is set as tunnel source",
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"tunnel_count": {
+										Description: "Number of tunnels per customer gateway. ",
+										Type:        schema.TypeInt,
+										Required:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"scale_group_id": {
+				Description: "The ID of the scale group associated with the connector.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"implicit_group_id": {
+				Description: "The ID of the implicit group associated with the connector.",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceConnectorGcpInterconnectCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	// INIT
+	client := m.(*alkira.AlkiraClient)
+	api := alkira.NewConnectorGcpInterconnect(m.(*alkira.AlkiraClient))
+
+	request, err := generateGcpInterconnectRequest(d, m)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// CREATE
+	response, provState, err, provErr := api.Create(request)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Set states
+	d.SetId(string(response.Id))
+
+	if client.Provision == true {
+		d.Set("provision_state", provState)
+
+		if provErr != nil {
+			return diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  "PROVISION (CREATE) FAILED",
+				Detail:   fmt.Sprintf("%s", provErr),
+			}}
+		}
+	}
+
+	return resourceConnectorGcpInterconnectRead(ctx, d, m)
+}
+
+func resourceConnectorGcpInterconnectRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	// INIT
+	client := m.(*alkira.AlkiraClient)
+	api := alkira.NewConnectorGcpInterconnect(m.(*alkira.AlkiraClient))
+
+	// GET
+	connector, provState, err := api.GetById(d.Id())
+
+	if err != nil {
+		return diag.Diagnostics{{
+			Severity: diag.Warning,
+			Summary:  "FAILED TO GET RESOURCE",
+			Detail:   fmt.Sprintf("%s", err),
+		}}
+	}
+
+	d.Set("name", connector.Name)
+	d.Set("description", connector.Description)
+	d.Set("cxp", connector.Cxp)
+	d.Set("group", connector.Group)
+	d.Set("size", connector.Size)
+	d.Set("tunnel_protocol", connector.TunnelProtocol)
+	d.Set("scale_group_id", connector.ScaleGroupId)
+	d.Set("billing_tag_ids", connector.BillingTags)
+	d.Set("loopback_prefixes", connector.LoopbackPrefixes)
+	d.Set("enabled", connector.Enabled)
+	d.Set("implicit_group_id", connector.ImplicitGroupId)
+	setGcpInterconnectInstance(d, connector, m)
+
```

Comment:
The segment options are a part of the instances in the API, I've just pulled them out to prevent having 4 layer nested config, the segmentOptions are set inside the `setGcpInterconnectInstance`.

<img width="1198" alt="image" src="https://github.com/user-attachments/assets/8beb5a83-f4b0-4f93-b184-86524fbac838">

Reply:
oh, haha 😮‍💨